Docs: Review documentation (#204)
aragon/connect: docs/advanced/app-connectors.md

> ### The connector

Reviewer: Should we add as an intro that we are describing the structure that we use in our own connectors (and that we recommend to follow)?

The `src` folder contains:

> - `models` folder: objects compatible with the `@aragon/connect` API.

Reviewer: ```suggestion
- `models` folder: objects compatible with the `@aragon/connect` API. This is what the consumers of your app connector are going to interact with.
```

> - `thegraph` folder: a connector implementation for The Graph which uses `@aragon/connect-thegraph` to perform GraphQL queries, parse them and present the result in the form of models objects. The `connector.ts` that has the app functions.

Reviewer: ```suggestion
- `thegraph` folder: a connector implementation for The Graph which uses `@aragon/connect-thegraph` to perform GraphQL queries, parse them and present the result in the form of models objects. This part is only used internally, and `connector.ts` contains the methods responsible to fetch the app data from The Graph.
```

Use `index.ts` to pick which models and objects are exposed to other packages.

> Use `connect.ts` to create the connector logic using `createAppConnector()` from `'@aragon/connect-core'`.

Reviewer: ```suggestion
Use `connect.ts` to create the connector logic [using `createAppConnector()`](#create-the-connector) from `'@aragon/connect-core'`.
```

> In `connector.ts`, add functions that the models of your connector will use. For example, the `Vote` model will call its `castsForVote(...): Promise<Cast[]>` function. These functions all follow the same structure of \(1\) performing a query, \(2\) parsing the results of a query, and \(3\) wrapping and returning the results in the appropriate model.

Reviewer: ```suggestion
In `thegraph/connector.ts` (assuming your connector is using The Graph), add functions that the models of your connector will use. For example, the `Vote` model of `@aragon/connect-voting` [will call its `castsForVote(...): Promise<Cast[]>` function](https://github.com/aragon/connect/blob/12dec7e5147220d29fb960bff01ff95e9ccca1bf/packages/connect-voting/src/models/Vote.ts#L39). These functions all follow the same structure of \(1\) performing a query, \(2\) parsing the results of a query, and \(3\) wrapping and returning the results in the appropriate model.
```

From the point of view of a consumer of your connector, the resulting object will be a function that can get used to connect an app.

> For example, this is how app authors can initiate a connection to the Voting app. In this example, `connectVoting` got returned by the implementation of `createAppConnector()` of the Voting connector:

Reviewer: I’m not sure I understand this, the implementation of `createAppConnector()` is not in the `Voting` connector? :thinking:

Author: You are right, I tried to clarify and I complicate it more 😅

#### appConnect\(app, connector\)

> The function returned by `createAppConnector()`, called by app authors. Takes these parameters:

Reviewer: ```suggestion
The function returned by `createAppConnector()`, called by app authors. It takes these parameters:
```
